cellengine/resources/gate.py

```diff
@@ -126,10 +132,30 @@ def update_gate_family(experiment_id: str, gid: str, body: Dict) -> None:
         if res["nModified"] < 1:
             raise Warning("No gates updated.")
 
-    def tailor_to(self, fcs_file: FcsFile):
-        self.tailored_per_file = True
-        self.fcs_file_id = fcs_file._id
-        self.update()
+    def apply_tailoring(self, fcs_file_ids: List[str]):
+        """Tailor this gate to a specific FCS file or files."""
+        res = ce.APIClient().apply_tailoring(self.experiment_id, self, fcs_file_ids)
+        return self._make_tailored_gates(res)
+
+    def _synthesize_gate(
+        self,
+        payload: Union[
+            ApplyTailoringInsert, ApplyTailoringUpdate, ApplyTailoringDelete
+        ],
+    ):
+        gate = self.to_dict()
+        gate.update(payload)
+        return Gate.from_dict(gate)
+
+    def _make_tailored_gates(self, payload: ApplyTailoringRes) -> Dict[str, List[Gate]]:
+        ret = defaultdict(list)
+        for k, v in payload.items():
+            if v:
+                if k == "deleted":
+                    [ret[k].append(i["_id"]) for i in v]
+                else:
+                    [ret[k].append(self._synthesize_gate(i)) for i in v]
+        return dict(ret)
```

Comment:
Nit: this is small enough now that you could inline it into `apply_tailoring` (and give `apply_tailoring` a return type).

Reply:
:+1: